cartoframes/utils/columns.py

```diff
@@ -100,115 +99,58 @@ def _extract_pgtype(fields):
-            self.database_type = self._db_column_type(geom_column, geom_type, dtype)
+    def __init__(self, name, dtype=None, geom_type=None):
+        self.name = name
+        self.dbname = normalize_name(name)
+        if str(dtype) == 'geometry':
+            self.is_geom = True
+            self.dbtype = 'geometry({}, 4326)'.format(geom_type or 'Point')
         else:
-            self.dataframe = None
-            self.database = Column.NORMALIZED_GEOM_COL_NAME
-            self.database_type = 'geometry(Point, 4326)'
+            self.is_geom = False
+            self.dbtype = dtypes2pg(dtype)
 
     def __repr__(self):
-        return "DataframeColumnInfo<{}, {}, {}>".format(
-            self.dataframe,
-            self.database,
-            self.database_type
+        return '{} {} {}'.format(
+            self.name,
+            self.dbname,
+            self.dbtype
         )
 
-    def _database_column_name(self, geom_column):
-        if geom_column and self.dataframe == geom_column:
-            normalized_name = Column.NORMALIZED_GEOM_COL_NAME
-        else:
-            normalized_name = normalize_name(self.dataframe)
-
-        return normalized_name
-
-    def _db_column_type(self, geom_column, geom_type, dtype):
-        if geom_column and self.dataframe == geom_column:
-            db_type = 'geometry({}, 4326)'.format(geom_type or 'Point')
-        else:
-            db_type = dtypes2pg(dtype)
-
-        return db_type
-
-    def __eq__(self, obj):
-        if isinstance(obj, dict):
-            return self.dataframe == obj['dataframe'] and \
-                self.database == obj['database'] and \
-                self.database_type == obj['database_type']
-        else:
-            return self.dataframe == obj.dataframe and \
-                self.database == obj.database and \
-                self.database_type == obj.database_type
-
 
 class DataframeColumnsInfo(object):
-    def __init__(self, df, with_lnglat=None):
-        self.df = df
-        self.with_lnglat = with_lnglat
-
-        self.geom_column = self._get_geom_col_name()
-        geom_type, enc_type = self._get_geometry_type()
-        self.geom_type = geom_type
-        self.enc_type = enc_type
-
-        self.columns = self._get_columns_info()
+    def __init__(self, df):
+        geom_column = self._get_geom_col_name(df)
+        geom_type = self._get_geometry_type(df, geom_column)
+        self.columns = self._get_columns_info(df, geom_column, geom_type)
 
     def __repr__(self):
         return str(self.columns)
 
-    def _get_columns_info(self):
-        df_columns = [(name, self.df.dtypes[name]) for name in self.df.columns]
-        if self.df.index.name is not None and self.df.index.name not in self.df:
-            df_columns.append((self.df.index.name, self.df.index.dtype))
+    def _get_geom_col_name(self, df):
+        geom_col = getattr(df, '_geometry_column_name', None)
+        if geom_col is None:
```

Comment:
If we try to upload a dataframe without geometry, it will raise, right?

Reply:
Yes. We need to remove this exception.

Reply:
https://github.com/CartoDB/cartoframes/pull/1265/commits/8681f47ac3129d8751a8a2a44350eaea563fa95b